src/test/scala/is/hail/expr/ir/ExtractAggregatorsSuite.scala

```diff
@@ -1,255 +1,108 @@
-        Sum(), Seq()),
-      addBoxedArray[java.lang.Double](region, 1.0, 10.0, null),
-      _ => 10.0)
-    assert(sum === 110.0)
-  }
 
   @Test
-  def filter1() {
-    val tAgg = TAggregable(TFloat64(), Map("scope0" -> (0, TFloat64())))
-    val region = Region()
-    val ir =
+  def fraction() {
+    val aggSig = AggSignature(Fraction(), TBoolean(), FastSeq())
+    assertEvalsTo(
       ApplyAggOp(
-        AggMap(
-          AggFilter(AggIn(tAgg), "x", ApplyBinaryPrimOp
-            (NEQ(), Ref("x", TFloat64()), F64(10.0))), "x",
-          ApplyBinaryPrimOp(Multiply(), Ref("scope0", TFloat64()), Ref("x", TFloat64()))),
-        Sum(), Seq())
-    val sum = run[Double, Double, Double](region,
-      ir,
-      addBoxedArray[java.lang.Double](region, 1.0, 10.0, null),
-      _ => 10.0)
-    assert(sum === 10.0)
+        SeqOp(Ref("a", TBoolean()), I32(0), aggSig),
+        FastSeq(), aggSig),
+      (FastIndexedSeq(Row(true), Row(false), Row(null), Row(true), Row(false)), TStruct("a" -> TBoolean())),
+      2.0 / 5.0)
   }
 
   @Test
-  def filter2() {
-    val tAgg = TAggregable(TFloat64(), Map("scope0" -> (0, TFloat64())))
-    val region = Region()
-    val ir =
+  def collectBoolean() {
+    val aggSig = AggSignature(Collect(), TBoolean(), FastSeq())
+    assertEvalsTo(
       ApplyAggOp(
-        AggMap(
-          AggFilter(AggIn(tAgg),
-            "x",
-            ApplyBinaryPrimOp(EQ(), Ref("x", TFloat64()), F64(10.0))),
-          "x",
-          ApplyBinaryPrimOp(Multiply(), Ref("scope0", TFloat64()), Ref("x", TFloat64()))),
-        Sum(), Seq())
-    val sum = run[Double, Double, Double](region,
-      ir,
-      addBoxedArray[java.lang.Double](region, 1.0, 10.0, null),
-      _ => 10.0)
-    assert(sum === 100.0)
+        SeqOp(Ref("a", TBoolean()), I32(0), aggSig),
+        FastSeq(), aggSig),
+      (FastIndexedSeq(Row(true), Row(false), Row(null), Row(true), Row(false)), TStruct("a" -> TBoolean())),
+      FastIndexedSeq(true, false, null, true, false))
   }
 
   @Test
-  def filter3() {
-    val tAgg = TAggregable(TFloat64(), Map("scope0" -> (0, TFloat64())))
-    val region = Region()
-    val ir =
+  def collectInt() {
+    val aggSig = AggSignature(Collect(), TInt32(), FastSeq())
+    assertEvalsTo(
       ApplyAggOp(
-        AggFilter(
-          AggMap(AggIn(tAgg),
-            "x",
-            ApplyBinaryPrimOp(Multiply(), Ref("scope0", TFloat64()), Ref("x", TFloat64()))),
-          "x",
-          ApplyBinaryPrimOp(EQ(), Ref("x", TFloat64()), F64(100.0))),
-        Sum(), Seq())
-    val sum = run[Double, Double, Double](region,
-      ir,
-      addBoxedArray[java.lang.Double](region, 1.0, 10.0, null),
-      _ => 10.0)
-    assert(sum === 100.0)
+        SeqOp(Ref("a", TInt32()), I32(0), aggSig),
+        FastSeq(), aggSig),
+      (FastIndexedSeq(Row(10), Row(null), Row(5)), TStruct("a" -> TInt32())),
+      FastIndexedSeq(10, null, 5))
   }
 
+  /*
+
```

Comment:
I want to see a working `flatMap` test before I approve this. I assume you'll convert the rest of these.